### ISSUE TITLE
ENOENT: no such file or directory, src/app/app.module.ts'

### ISSUE THREAD

**Issue by albanx** [angular/angular-cli#1695]
1. OS?  Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)
MAC yosemite

2. Versions. Please run `ng --version`. If there's nothing outputted, please run
in a Terminal: `node --version` and paste the result here:
```
angular-cli: 1.0.0-beta.11-webpack.2
node: 4.4.7
os: darwin x64
alban-mac:public
```

3. Repro steps. Was this an app that wasn't created using the CLI? What change did you
   do on your code? etc.
```
ng g component test.component
```
Just trying to create a module with the webpack. To note that I have renamed my app.module to mytestapp.module

Cheers

**Comment by filipesilva** [comment]
Did you also change the import in `main.ts` to use your renamed `mytestapp.module` file?

**Comment by albanx** [comment]
@filipesilva  Yes I did.

**Comment by monojack** [comment]
Are you importing it somewhere else using `app.module`? Maybe inside a barrel or something?

**Comment by albanx** [comment]
@monojack  I made a search in the code, and I cannot find ``app.module``, the application is working fine, and the cli works fine, except for the fact that it complains about ``app.module``:

```
$ ng g component shared/components/lang-selector.component
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
installing component
  create src/app/shared/components/lang-selector/lang-selector.component.css
  create src/app/shared/components/lang-selector/lang-selector.component.html
  create src/app/shared/components/lang-selector/lang-selector.component.spec.ts
  create src/app/shared/components/lang-selector/lang-selector.component.ts
  create src/app/shared/components/lang-selector/index.ts
ENOENT: no such file or directory, open '/export/www/alban/mytestapp/public_web/src/app/app.module.ts'
Error: ENOENT: no such file or directory, open '/export/www/alban/mytestapp/public_web/src/app/app.module.ts'
    at Error (native)
    at Object.fs.openSync (fs.js:549:18)
    at Object.fs.readFileSync (fs.js:397:15)
    at getSource (/export/www/alban/mytestapp/public_web/node_modules/angular-cli/addon/ng2/utilities/ast-utils.ts:24:45)
    at _addSymbolToNgModuleMetadata (/export/www/alban/mytestapp/public_web/node_modules/angular-cli/addon/ng2/utilities/ast-utils.ts:137:18)
    at Object.addComponentToModule (/export/www/alban/mytestapp/public_web/node_modules/angular-cli/addon/ng2/utilities/ast-utils.ts:235:12)
    at Class.module.exports.files.fileList.afterInstall.options [as afterInstall] (/export/www/alban/mytestapp/public_web/node_modules/angular-cli/addon/ng2/blueprints/component/index.js:138:18)
    at lib$rsvp$$internal$$tryCatch (/export/www/alban/mytestapp/public_web/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/export/www/alban/mytestapp/public_web/node_modules/rsvp/dist/rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (/export/www/alban/mytestapp/public_web/node_modules/rsvp/dist/rsvp.js:1019:11)
    at lib$rsvp$asap$$flush (/export/www/alban/mytestapp/public_web/node_modules/rsvp/dist/rsvp.js:1198:9)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)
```

**Comment by filipesilva** [comment]
I think I know what the issue is. @hansl @Brocco currently the `addComponentToModule` functionality seems to be assuming `app.module.ts`. This needs a module discovery mechanism instead of a hardcoded file.

**Comment by Brocco** [comment]
This is something that I will be working on soon, once I finish another related feature.

**Comment by Brocco** [comment]
This issue was resolved by PR #1966, closing
